[5.4] add schedule:list command

jaumesala opened the pull request:
As discused on #20555 this command shows basic information of the scheduled tasks. The expresion string is written "as is" in the event class, so the minutes are not correlated with the cron expression (as @laurencei pointed out). Still I think this command can be helpful to developers looking for a fast way to see what's being sent to the task scheduler. 

![schedule-list](https://user-images.githubusercontent.com/385551/29315517-e81b56cc-81c3-11e7-9e1f-67b5d6b8aabd.png)

jaumesala committed:
add schedule:list command

jaumesala committed:
Fix StyleCI

decadence commented:
Related: https://github.com/laravel/framework/pull/19811

jaumesala committed:
Fix StyleCI

laurencei commented on src/Illuminate/Console/Scheduling/ScheduleListCommand.php:
```diff
@@ -0,0 +1,178 @@
+        $events = $this->getEvents();
+
+        if (empty($events)) {
+            return $this->info('Your application has no scheduled commands to run.');
```
Maybe different message? "There are no scheduled commands to list" or something.

The error message sounds like it's trying to run them (which it is not)

jaumesala replied:
Yes, You are right.

taylorotwell commented:
I still feel like the simplest way to view your schedule is to simply look at your schedule calls in the Console kernel itself.

jaumesala commented:
I don't know, I've been recently working on a project that kept me all the time opening the Console Kernel file, a command like this would have been very useful. Looks like @decadence was also trying to achieve the same results (I didn't see your pull request sorry).
What if we remove the *expression* column and focus on listing what's in the Console Kernel?
We can change the *expression* for *next execution* like @decadence did in his pull request.

decadence commented:
I still think this command will be useful for quick view on schedule via console.
@jaumesala your PR more informative and we could add my "Next Execution date" column to it.
Unfortunately @taylorotwell has another opinion.

jaumesala commented:
Even if this PR is closed, I've just updated my branch with the discussed points. I have added @decadence idea and corrected the empty message as suggested by @laurencei. 

![schedule-list2](https://user-images.githubusercontent.com/385551/29338066-04312602-8214-11e7-997d-ddd880f802be.png)

decadence commented:
@jaumesala thanks. I will save it for the future use ;)

RikSomers commented:
@taylorotwell To be honest, the kernel file can sometimes get rather big and complicated, so having a command to show which tasks are scheduled can be extremely useful in my opinion.

laurencei commented:
This is a perfect snipped of code that can be easily wrapped in a package.

In fact - I quickly googled and one already exists: https://github.com/hmazter/laravel-schedule-list

So if you want the functionality - your best bet is just install the package...

thecrypticace commented:
Heh, I was just looking for something like this yesterday to verify that something was indeed running at the time I thought it was supposed to. Thought Laravel included it by default but at least there's a package for it. 👍
